### ISSUE TITLE
Metadata for locale 

### ISSUE THREAD

**Issue by lcharette** [userfrosting/UserFrosting#850]
See https://github.com/userfrosting/i18n/issues/4

This is dumb, as it doesn't works if the `default` has a space somewhere : https://github.com/userfrosting/UserFrosting/blob/develop/app/sprinkles/core/src/ServicesProvider/ServicesProvider.php#L437

This is rubbish and that's not how the default works : https://github.com/userfrosting/UserFrosting/blob/develop/app/sprinkles/core/src/ServicesProvider/ServicesProvider.php#L451

And that whole thing is just messy : https://github.com/userfrosting/UserFrosting/blob/develop/app/sprinkles/core/config/default.php#L140-L159

**tl;dr** FU the current `available` config, FU the `default` config with it's obscure order. Locale metadata with dependencies and name FTW. An array of available locale FTW.

**Comment by lcharette** [comment]
So first question. How do we store locales config and where ?

In a Yaml file? In `/locale/fr_FR/fr_FR.yaml` or in `locales/fr_FR.yaml` ? In core sprinkle? A new sprinkle add a new locale / wants to replace the config they replace said file in their own `locales/` folder ?

**Comment by lcharette** [comment]
This as been implemented with new version of i18n for UF 4.4
